app/helpers/heatmap_helper.rb

```diff
@@ -84,7 +84,7 @@ def self.sample_taxons_dict(params, samples, background_id)
 
     unless taxon_ids.empty?
       # Refetch at genus level using species level
-      parent_ids = species_selected ? [] : HeatmapHelper.fetch_parent_ids(taxon_ids, samples)
```

Comment:
notice how `taxon_ids` was plucked from `details` then used to fetch `parents_ids`. instead we can include the parent ids in `details`.